graph/telemetry/istio/appender/service_entry.go

```diff
@@ -72,14 +72,46 @@ func (a ServiceEntryAppender) getServiceEntry(service string, globalInfo *graph.
 						location = "MESH_INTERNAL"
 					}
 					for _, host := range entry.Spec.Hosts.([]interface{}) {
-						serviceEntries[host.(string)] = location
+						serviceEntryHosts = append(serviceEntryHosts, serviceEntryHost{
+							location: location,
+							host:     host.(string),
+						})
 					}
 				}
 			}
 		}
-		log.Tracef("Found [%v] service entries", len(serviceEntries))
+		globalInfo.Vendor[serviceEntryHostsKey] = serviceEntryHosts
 	}
 
-	location, ok := serviceEntries[service]
-	return location, ok
+	for _, serviceEntryHost := range serviceEntryHosts {
+		// handle exact match
+		if serviceEntryHost.host == serviceName {
+			return serviceEntryHost.location, true
+		}
+		// handle serviceName prefix (e.g. host = serviceName.namespace.svc.cluster.local)
+		if serviceEntryHost.location == "MESH_INTERNAL" {
+			if strings.Split(serviceEntryHost.host, ".")[0] == serviceName {
+				return serviceEntryHost.location, true
+			}
+		}
+		// handle wildcard
+		if serviceEntryHost.location == "MESH_EXTERNAL" {
+			hostTokens := strings.Split(serviceEntryHost.host, ".")
+			serviceNameTokens := strings.Split(serviceName, ".")
+			match := len(hostTokens) == len(serviceNameTokens)
+			if match {
+				for i, t := range hostTokens {
+					if t != "*" && t != serviceNameTokens[i] {
```

Comment:
Could a wildcard also be used for MESH_INTERNAL?
`*.my-namespace.*.*.*`

Reply:
Not as far as I know: see the "hosts" description [here](https://istio.io/docs/reference/config/networking/v1alpha3/service-entry/#ServiceEntry).

For MESH_INTERNAL I'm only trying to support exact match, or to match the serviceName portion of a full service.